test/lib/jdk/test/lib/hotspot/ir_framework/README.md

```diff
@@ -0,0 +1,137 @@
+3. After the compilation, the test is invoked one more time.
+
+Once the test VM terminates, IR verification (if possible) is performed on the output of the test VM. If any test throws an exception during its execution or if IR matching fails, the failures are collected and reported in a pretty format. Check the standard error and output for more information and how to reproduce these failures.
+
+Some of the steps above can be different due to the kind of the test or due to using non-default annotation properties. These details and differences are described in the Javadocs for the three tests (see section 2.1 Different Tests).
+
+More information about the internals and the workflow of the framework can be found in the Javadocs of [TestFramework](./TestFramework.java).  
+ 
+## 4. Internal Framework Tests
+There are various tests to verify the correctness of the test framework. These tests can be found in [tests](tests) and can directly be run with JTreg. The tests are not part of the normal JTreg tests of HotSpot and should only be run upon changing the framework code as a minimal form of testing.
+
+Additional testing was performed by converting all compiler Inline Types tests that used the currently present IR test framework in Valhalla (see [JDK-8263024](https://bugs.openjdk.java.net/browse/JDK-8263024)). It is strongly advised to make sure a change to the framework still lets these converted tests in Valhalla pass as part of an additional testing step.
+
+## 5. Future Work
+This framework is based on the idea of the currently present IR test framework in [Valhalla](https://github.com/openjdk/valhalla/blob/e9c78ce4fcfd01361c35883e0d68f9ae5a80d079/test/hotspot/jtreg/runtime/valhalla/inlinetypes/InlineTypesTest.java). This IR framework was used with great success in Valhalla and thus served as a foundation for this new IR framework.
+ 
+ The new framework supports all the features that are present in the Valhalla IR framework with the idea to replace it at some point. The initial design and feature set was kept simple and straight forward and serves well for small to medium sized tests. There are a lot of possibilities to further enhance the framework and make it more powerful. This can be tackled in additional RFEs. A few ideas include:
+
+- Provide more default IR regexes (e.g. for vector nodes etc.) in [IRNode](./IRNode.java) ([JDK-8265197](https://bugs.openjdk.java.net/browse/JDK-8265197)).
+- Parsing the IR directly to perform queries on it, walking it, searching it, match on IR node properties etc. ([JDK-8265198](https://bugs.openjdk.java.net/browse/JDK-8265198)).
+- Investigate more verification possibilities based on additional/other flags (e.g. _-XX:+PrintIdealGraph_, _-XX:+TraceNewVectors_ etc.) ([JDK-8265200](https://bugs.openjdk.java.net/browse/JDK-8265200)).
+- More interface methods for verification in checked and custom run tests.
+- More stress/debug framework flags.
+- Additional check possibilities in `@IR` annotations. 
```

Comment:
I would remove the "Future Work" section and migrate it completely to RFEs. Otherwise it will quickly be outdated anyway.